### PR TITLE
Rename pool limit options

### LINKED ISSUE #948: Deprecate ProxyLimits naming in favour of max_connections and max_keepalive
Prompted by https://github.com/encode/httpx/pull/942#discussion_r424341967

The `PoolLimits` class uses `soft_limit` and `hard_limit` whereas in httpcore we use `max_connections` and `max_keepalive` which are clearer.

### PR DESCRIPTION
**Pull request by tomchristie**
Closes #948

### CODE REVIEW

**Comment by tomchristie** [comment]
The URLLib3 changes here will become redundant after #966, but I'm going to ignore that here to keep this PR tightly scoped.

File: httpx/_config.py [hunk 1]
```diff
@@ -288,29 +289,47 @@ class PoolLimits:
...
+            warnings.warn(
+                "'soft_limit' is deprecated. Use 'max_keepalive' instead.",
+                DeprecationWarning,
+            )
+        if hard_limit is not None:  # pragma: nocover
+            self.max_connections = hard_limit
+            warnings.warn(
+                "'hard_limit' is deprecated. Use 'max_connections' instead.",
+                DeprecationWarning,
+            )
```
Comment (yeraydiazdiaz) [inline httpx/_config.py:319]: You can use the new [`warn_deprecated` util function.](https://github.com/encode/httpx/blob/master/httpx/_utils.py#L406-L407)

File: httpx/_config.py [hunk 2]
```diff
@@ -288,29 +288,43 @@ class PoolLimits:
...
+        self.max_keepalive = max_keepalive
+        self.max_connections = max_connections
+        if soft_limit is not None:  # pragma: nocover
+            self.max_keepalive = soft_limit
+            warn_deprecated("'soft_limit' is deprecated. Use 'max_keepalive' instead.",)
+        if hard_limit is not None:  # pragma: nocover
+            self.max_connections = hard_limit
+            warn_deprecated(
+                "'hard_limit' is deprecated. Use 'max_connections' instead.",
+            )
```
Comment (florimondmanca) [inline httpx/_config.py:314]: I _assume_ eventually one of the linters is going to cough on the extra commas here, so just a heads up:

```suggestion
        if soft_limit is not None:  # pragma: nocover
            self.max_keepalive = soft_limit
            warn_deprecated("'soft_limit' is deprecated. Use 'max_keepalive' instead.")
        if hard_limit is not None:  # pragma: nocover
            self.max_connections = hard_limit
            warn_deprecated(
                "'hard_limit' is deprecated. Use 'max_connections' instead."
            )
```

**Comment by tomchristie** [comment]
Merged #967 into this as we're dependant on it.

File: httpx/_config.py (hunk 2 above)
Comment (florimondmanca) [reply httpx/_config.py:314]: Actually not, feel free to ignore!

**Pull request merged**